### PR TITLE
`Development`: Extract client duplicates into re-usable components

### PR DESCRIPTION
**Pull request by aplr**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [ ] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://ls1intum.github.io/Artemis/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://ls1intum.github.io/Artemis/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I followed the [coding and design guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

While implementing new features in Artemis, components have been duplicated. This PR extracts those duplicate components into reusable ones and replaces the original implementations.

### Description

The following components are affected:

*confirm-entity-name*

* Replaced inline-implementation in `delete-dialog` component
* Replaced inline-implementation in `data-export-confirmation` component
* Replaced inline-implementation in `reset-repo-button` component

*working-time-control*

* Replaced the inline-implementation in student-exam-detail component

#### Delete dialog

Prerequisites:
- 1 Instructor
- Some resource to delete

1. Log in to Artemis
2. Navigate to Course Administration
3. Open a course, exam, ...
4. Check if the delete dialog still works

#### Exam working time change

Prerequisites:
- 1 Instructor
- 1 Exam

1. Log in to Artemis
2. Navigate to Exam management
3. Check if you can still edit the working time w/ the working time dialog

#### Student exam working time change

Prerequisites:
- 1 Instructor
- 1 Exam

1. Log in to Artemis
2. Navigate to Exam management
3. Open the student exam details
4. Check if you can still edit the individual student exam working time, both absolute and relative

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots

(no changes, just refactoring)

### CODE REVIEW

File: src/test/javascript/spec/component/legal/data-export-confirmation-dialog.component.spec.ts [hunk 1]
```diff
@@ -55,26 +59,32 @@ describe('DataExportConfirmationDialogComponent', () => {
...
         // Form can't be submitted if the expected login doesn't match the entered login
         comp.expectedLogin = 'login';
         comp.enteredLogin = '';
         comp.dialogError = new Observable<string>();
         fixture.detectChanges();
-        let submitButton = debugElement.query(By.css('.btn.btn-primary'));
-        expect(submitButton.nativeElement.disabled).toBeTrue();
+        await fixture.whenStable();
+        expect(comp.dataExportConfirmationForm.invalid).toBeTrue();
+        // TODO: why do changes not propagate to the UI?
```
Comment (aplr) [inline src/test/javascript/spec/component/legal/data-export-confirmation-dialog.component.spec.ts:70]: Does sb. know why the form is `invalid`, but `disabled` is not `true` on the button? Somehow, it does not seem to render, but I don't know why.

**Comment by Strohgelaender** [comment]
Is this PR ready to review? It has the label but is still in draft state.

**Comment by aplr** [comment]
@Strohgelaender yes it is forgot to put it out of draft mode

File: src/test/javascript/spec/component/legal/data-export-confirmation-dialog.component.spec.ts (hunk 1 above)
Comment (MaximilianAnzinger) [reply src/test/javascript/spec/component/legal/data-export-confirmation-dialog.component.spec.ts:70]: we usually use `fakeAsync`

**Review by milljoniaer — DISMISSED** [review]
Tested on ts4
- Course, Exam, Exercise Creation and Deletion
- Exam Time change in general and for specific students

works as expected

**Review by milljoniaer — DISMISSED** [review]
re-approval 👍

**Review by reschandreas — DISMISSED** [review]
demo in testing session, lgtm

**Review by laurenzfb — DISMISSED** [review]
tested in testing session

**Review by valentin-boehm — DISMISSED** [review]
lgtm

**Review by krusche — CHANGES_REQUESTED** [review]
e2e tests fail and many checkboxes are not filled, this PR is not ready to merge

**Comment by xHadie** [comment]
_Tested on TS3 Legacy_
**Delete dialog**
Deletion works but:
Delete button for exams (Page: Exam Management) is missing
image
Steps to reproduce:
1. Click on Course Management in the header
2. Select any course that has an exam
3. Click on **Edit** (perhaps a delete button is missing here too?)
image
4. There will be no delete button on the exam edit view. Whole button group appears to be missing:
image

**Exam working time change**
LGTM

**Student exam working time change**
LGTM

**Comment by JanaNF** [comment]
Tested on ts3
1. Delete dialog: Worked successfully
2. Exam working time change: Worked successfully
3. Student exam working time change: Worked successfully

**Review by milljoniaer — DISMISSED** [review]
Tested on ts4, works as expected

**Review by reschandreas — APPROVED** [review]
retested on ts4, still LGTM

**Review by JanaNF — DISMISSED** [review]
Tested on ts3

Delete dialog: Worked successfully
Exam working time change: Worked successfully
Student exam working time change: Worked successfully

**Review by valentin-boehm — APPROVED** [review]
still lgtm

**Review by reschandreas — APPROVED** [review]
still LGTM

**Review by valentin-boehm — DISMISSED** [review]
reapprove after merge

**Review by Gusti2010 — DISMISSED** [review]
Tested manually on Ts2_Legacy and worked as expected! 
Ts11_Legacy could not create exams for the students. However this is not related to this PR.

**Review by reschandreas — APPROVED** [review]
reapprove after code change

**Pull request merged**